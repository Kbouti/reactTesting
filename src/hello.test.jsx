import { describe, it, expect } from 'vitest';

// describe(`something truthy and falsy`, () => {
//     it(`true to be true`, () => {
//         expect(true).toBe(true);
//     });

//     it(`false to be false`, () => {
//         expect(false).toBe(false);
//     })
// })



import {render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Hello from "./hello.jsx";

describe("Hello component", () => {
    it("renders Magnificent Monkeys", () => {
          // since screen does not have the container property, we'll destructure render to obtain a container for this test
        const {container} = render(<Hello/>)
        expect(container).toMatchSnapshot();
    });


it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();

    render(<Hello/>);
    const button = screen.getByRole("button", {name: "Click me!"});

    await user.click(button);
    expect(screen.getByRole("heading").textContent).toMatch(/Radical Rhinos/i);
}

)})