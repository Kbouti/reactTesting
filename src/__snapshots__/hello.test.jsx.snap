// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello component > renders Magnificent Monkeys 1`] = `
<div>
  <button
    type="button"
  >
    Click me! 
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;
